app/controllers/CheckDetailsController.scala

```diff
@@ -125,79 +124,93 @@ class CheckDetailsController @Inject()(
                                                mode: Mode,
                                                sessionWithNINO: Session,
                                                origin: Option[String],
-                                               pdvData: PDVResponseData)(implicit headerCarrier: HeaderCarrier): Result = {
-    if (pdvData.validationStatus.equals("")) {
-      Redirect(routes.InvalidDataNINOHelpController.onPageLoad(mode = mode)).withSession(sessionWithNINO)
-    } else if (pdvData.validationStatus.equals("failure")) {
-      logger.warn(s"PDV matched failed: ${pdvData.validationStatus}")
-      auditService.findYourNinoPDVMatchFailed(pdvData, origin)
-      Redirect(routes.InvalidDataNINOHelpController.onPageLoad(mode = mode)).withSession(sessionWithNINO)
-    } else {
-      val errorStatusCode: Option[String] = idDataError match {
-        case conError: ConnectorError => Some(conError.statusCode.toString)
-        case _ => None
-      }
+                                               pdvData: PDVResponseData
+                                              )(implicit hc: HeaderCarrier, request: DataRequest[AnyContent]): Result =
+    pdvData.validationStatus match {
+      case "failure" =>
+        logger.warn(s"PDV matched failed: ${pdvData.validationStatus}")
+        auditService.findYourNinoPDVMatchFailed(pdvData, origin)
+        redirectToInvalidDataNINOHelpController(mode, sessionWithNINO)
+      case "success" =>
+        handleSuccessValidationStatus(idDataError, origin, pdvData)
+      case _ =>
+        checkDetailsMatchingFailedWithUnknownIssue(mode)
+    }
+
+  private def handleSuccessValidationStatus(idDataError: IndividualDetailsError,
+                                            origin: Option[String],
+                                            pdvData: PDVResponseData
+                                           )(implicit hc: HeaderCarrier, request: DataRequest[AnyContent]): Result = {
+    auditService.findYourNinoPDVMatched(pdvData, origin, None)
 
-      auditService.findYourNinoIdDataError(pdvData, errorStatusCode, idDataError, origin)
-      logger.warn(s"Failed to retrieve Individual Details data: ${idDataError.errorMessage}")
-      Redirect(routes.InvalidDataNINOHelpController.onPageLoad(mode = mode)).withSession(sessionWithNINO)
+    val errorStatusCode: Option[String] = idDataError match {
+      case conError: ConnectorError => Some(conError.statusCode.toString)
+      case _ => None
     }
+
+    auditService.findYourNinoIdDataError(pdvData, errorStatusCode, idDataError, origin)
+    logger.warn(s"Failed to retrieve Individual Details data: ${idDataError.errorMessage}")
+    ServiceUnavailable(errorHandler.standardErrorTemplate())
```

Comment:
Why specifically a 503? This feels wrong.

Reply:
we have not decided on this. I just gave that 503 because most of the API#1694 errors are 503 in kibana logs.